tracer/src/Datadog.Trace.Manual/Configuration/TracerSettings.cs

```diff
@@ -0,0 +1,546 @@
+    private OverrideValue<Uri> _agentUriOverride = new();
+    private List<int>? _httpClientErrorCodes;
+    private List<int>? _httpServerErrorCodes;
+    private Dictionary<string, string>? _serviceNameMappings;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TracerSettings"/> class with default values.
+    /// </summary>
+    [Instrumented]
+    public TracerSettings()
+        : this(PopulateDictionary(new(), useDefaultSources: false), isFromDefaultSources: false)
+    {
+        // TODO: _Currently_ this doesn't load _any_ configuration, so it feels like an error for customers to use it?
+        // I'm wondering if we should _always_ populate from the default sources instead, as otherwise seems
+        // like an obvious point of confusion?
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TracerSettings"/> class with default values,
+    /// or initializes the configuration from environment variables and configuration files.
+    /// Calling <c>new TracerSettings(true)</c> is equivalent to calling <c>TracerSettings.FromDefaultSources()</c>
+    /// </summary>
+    /// <param name="useDefaultSources">If <c>true</c>, creates a <see cref="TracerSettings"/> populated from
+    /// the default sources such as environment variables etc. If <c>false</c>, uses the default values.</param>
+    [Instrumented]
+    public TracerSettings(bool useDefaultSources)
+        : this(PopulateDictionary(new(), useDefaultSources), useDefaultSources)
+    {
+    }
+
+    // Internal for testing
+    internal TracerSettings(Dictionary<string, object?> initialValues, bool isFromDefaultSources)
+    {
+        // The values set here represent the defaults when there's no auto-instrumentation
+        // We don't care too much if they get out of sync because that's not supported anyway
+        _agentUriInitial = GetValue<Uri?>(initialValues, TracerSettingKeyConstants.AgentUriKey, null) ?? new Uri("http://127.0.0.1:8126");
+        _analyticsEnabledInitial = GetValue(initialValues, TracerSettingKeyConstants.AnalyticsEnabledKey, false);
+        _customSamplingRulesInitial = GetValue<string?>(initialValues, TracerSettingKeyConstants.CustomSamplingRules, null);
+        _disabledIntegrationNamesInitial = GetValue<HashSet<string>?>(initialValues, TracerSettingKeyConstants.DisabledIntegrationNamesKey, null) ?? [];
+        _diagnosticSourceEnabledInitial = GetValue(initialValues, TracerSettingKeyConstants.DiagnosticSourceEnabledKey, false);
+        _environmentInitial = GetValue<string?>(initialValues, TracerSettingKeyConstants.EnvironmentKey, null);
+        _globalSamplingRateInitial = GetValue<double?>(initialValues, TracerSettingKeyConstants.GlobalSamplingRateKey, null);
+        _globalTagsInitial = GetValue<IDictionary<string, string>?>(initialValues, TracerSettingKeyConstants.GlobalTagsKey, null) ?? new ConcurrentDictionary<string, string>();
```

Comment:
NIT: it could be nice to have an empty const readonly version of the dictionary for these cases, to avoid the extra instances

Reply:
This only happens in manual-only, so going to ignore that for now for simplicity